File: doc-examples/jdbc-example-java/src/main/java/example/TeacherSubView.java
package example;

import io.micronaut.data.annotation.*;

@Embeddable
// tag::class-example[]
@JsonSubView(entity = Teacher.class)
public class TeacherSubView {
    @Id
    @MappedProperty(value = "id")
    private Long teachID;

    @MappedProperty(value = "name")
    private String teacher;
}
// end::class-example[]

File: doc-examples/jdbc-example-python/src/main/python/example/TeacherSubView.py
from dataclasses import dataclass
from typing import Annotated

from micronaut.data.annotation import Embeddable, Id, JsonSubView, MappedProperty

from example.Teacher import Teacher


@Embeddable
# tag::class-example[]
@JsonSubView(entity=Teacher)
@dataclass
class TeacherSubView:
    teachID: Annotated[int | None, Id, MappedProperty(value="id")] = None
    teacher: Annotated[str | None, MappedProperty(value="name")] = None
# end::class-example[]
